lib/tasks/data/export_workspace_data.rake

```diff
@@ -0,0 +1,92 @@
+# WORKSPACE_DATA_S3_BUCKET='s3-bucket-name' STATUSES_TO_EXPORT='False,Verified' bundle exec rake check:data:export_workspace_data[workspace-slug-1,workspace-slug-2,...,workspace-slug-N]
+
+namespace :check do
+  namespace :data do
+    desc 'Export all workspace data into a CSV uploaded to S3'
+    task export_workspace_data: :environment do |_t, params|
+      old_logger = ActiveRecord::Base.logger
+      ActiveRecord::Base.logger = nil
+
+      statuses_to_export = ENV.fetch('STATUSES_TO_EXPORT').to_s.split(',')
+      bucket_name = ENV.fetch('WORKSPACE_DATA_S3_BUCKET')
+      region = 'eu-west-1'
+      begin
+        s3_client = Aws::S3::Client.new(region: region)
+      rescue Aws::Sigv4::Errors::MissingCredentialsError
+        puts 'Please provide the AWS credentials.'
+        exit 1
+      end
+
+      def object_uploaded?(s3_client, bucket_name, object_key, file_path)
+        response = s3_client.put_object(
+          bucket: bucket_name,
+          key: object_key,
+          body: File.read(file_path)
+        )
+        if response.etag
+          return true
+        else
+          return false
+        end
+      rescue StandardError => e
+        puts "Error uploading S3 object: #{e.message}"
+        return false
+      end
+
+      slugs = params.to_a
+      slugs.each do |slug|
+        team = Team.find_by_slug(slug)
+        filename = "workspace-data-#{slug}-#{Time.now.strftime('%Y-%m-%d')}.csv"
+        filepath = File.join(Rails.root, 'tmp', filename)
+        puts "Exporting data for workspace and saving to #{filepath}..."
+        output = File.open(filepath, 'w+')
+
+        header = ['Claim', 'Status', 'Created by', 'Submitted at', 'Published at', 'Number of media', 'Media: Shares', 'Media: Reactions', 'Media: Comments', 'Tags', 'Reviewed by']
+        fields = team.team_tasks.sort
+        fields.each { |tt| header << tt.label }
+        output.puts(header.collect{ |x| '"' + x.to_s.gsub('"', '') + '"' }.join(','))
+
+        n = team.project_medias.count
+        i = 0
+        team.project_medias.find_each do |pm|
+          i += 1
+          puts "[#{i}/#{n}] Exporting item from workspace #{slug}..."
+          status = pm.status_i18n
+          if statuses_to_export.include?(status)
+            row = [
+              pm.claim_description&.description,
+              status,
+              pm.author_name,
+              pm.created_at.strftime("%Y-%m-%d %H:%M:%S"),
+              pm.published_at&.strftime("%Y-%m-%d %H:%M:%S"),
+              pm.linked_items_count(true),
```

Comment:
why you need to recalculate cached field? it'll be faster to call cached field without true

Reply:
Ah yes, I did it for testing... fixed now, good catch.